mlflow/tracking/client.py

```diff
@@ -1526,6 +1526,121 @@ def log_table(
+
+            client = MlflowClient()
+            loaded_table = client.load_table(
+                experiment_id="uuid-0",
+                artifact_file="qabot_eval_results.json",
+                # Append the run ID and the parent run ID to the table
+                extra_columns=["run_id", "parent_run_id"],
+            )
+        """
+        import pandas as pd
+
+        subset_tag_value = json.dumps({"path": artifact_file, "type": "table"})
+
+        # Build the filter string
+        filter_string = f"tags.{MLFLOW_LOGGED_ARTIFACTS} LIKE '%{subset_tag_value}%'"
+        if run_ids:
+            list_run_ids = ",".join(map(repr, run_ids))
+            filter_string += f" and attributes.run_id IN ({list_run_ids})"
+
+        runs = mlflow.search_runs(experiment_ids=[experiment_id], filter_string=filter_string)
+        if run_ids and len(run_ids) != len(runs):
+            _logger.warning(
+                "Not all runs have the specified table artifact. Some runs will be skipped."
+            )
+
+        # TODO: Add parallelism support here
+        def get_artifact_data(run):
+            run_id = run.run_id
+            norm_path = posixpath.normpath(artifact_file)
+            artifact_dir = posixpath.dirname(norm_path)
+            artifact_dir = None if artifact_dir == "" else artifact_dir
+            existing_predictions = pd.DataFrame()
+
+            artifacts = [
+                f.path for f in self.list_artifacts(run_id, path=artifact_dir) if not f.is_dir
+            ]
+            if artifact_file in artifacts:
+                with tempfile.TemporaryDirectory() as tmpdir:
+                    downloaded_artifact_path = mlflow.artifacts.download_artifacts(
+                        run_id=run_id, artifact_path=artifact_file, dst_path=tmpdir
+                    )
+                    existing_predictions = pd.read_json(downloaded_artifact_path, orient="split")
+                    if extra_columns is not None:
+                        for column in extra_columns:
+                            if column in existing_predictions:
+                                column_name = f"{column}_"
+                                _logger.warning(
+                                    f"Column name {column} already exists in the table. "
+                                    "Resolving the conflict, by appending an underscore "
+                                    "to the column name."
+                                )
+                            else:
+                                column_name = column
+                            existing_predictions[column_name] = run[column]
+
+            else:
+                raise MlflowException(f"Artifact {artifact_file} not found for run {run_id}.")
+
+            return existing_predictions
+
+        return (
+            pd.concat([get_artifact_data(run) for _, run in runs.iterrows()], ignore_index=True)
+            if not runs.empty
+            else pd.DataFrame()
```

Comment:
I think we should throw if the table doesn't exist. It seems a bit confusing to return an empty dataframe